[pt] Localize content/pt/docs/languages/python/getting-started.md (#6292)
open-telemetry/opentelemetry.io: content/pt/docs/languages/python/getting-started.md

---
title: Primeiros Passos

> description: Obtenha telemetria para seu aplicativo em menos de 5 minutos!

Reviewer: ```suggestion
description: Obtenha telemetria para sua aplicação em menos de 5 minutos!
```

weight: 10
# prettier-ignore
cSpell:ignore: debugexporter diceroller distro maxlen randint rolldice rollspan venv
default_lang_commit: 43e2cb3b4d0dd513b436add73236503a8d592b39
---

Esta página mostrará como começar a usar o OpenTelemetry em Python.

> Você aprenderá como instrumentar automaticamente um aplicativo simples, de forma

Reviewer: ```suggestion
Você aprenderá como instrumentar automaticamente uma aplicação simples, de forma
```

que [rastros][], [métricas][], e [logs][] sejam emitidos para o console.

## Pré-requisitos {#prerequisites}

Certifique-se de ter o seguinte instalado localmente:

- [Python 3](https://www.python.org/)

## Aplicação de Exemplo {#example-application}

> O exemplo a seguir usa um aplicativo básico

Reviewer: ```suggestion
O exemplo a seguir usa uma aplicação básica em
```

[Flask](https://flask.palletsprojects.com/). Se você não estiver usando Flask,

> tudo bem — você pode usar OpenTelemetry Python com outros frameworks web também,

Reviewer 2: ```suggestion
tudo bem — você pode usar OpenTelemetry Python com outros _frameworks_ web também,
```

> como Django e FastAPI. Para uma lista completa de bibliotecas para frameworks

Reviewer: ```suggestion
como Django e FastAPI. Para uma lista completa de bibliotecas para _frameworks_
```

> suportados, veja o

Reviewer: +1 eu gosto do consulte tbm

[registro](/ecosystem/registry/?component=instrumentation&language=python).

> Para exemplos mais elaborados, veja

Reviewer 2: Mesmo ponto do comentário anterior, caso faça sentido :) 

```suggestion
Para exemplos mais elaborados, consulte
```

[exemplos](/docs/languages/python/examples/).

## Instalação {#installation}

Para começar, configure um ambiente em um novo diretório:

```shell
mkdir otel-getting-started
cd otel-getting-started
python3 -m venv venv
source ./venv/bin/activate
```

Agora instale Flask:

```shell
pip install flask
```

### Crie e inicie um Servidor HTTP {#create-and-launch-an-http-server}

Crie um arquivo `app.py` e adicione o seguinte código a ele:

```python
from random import randint
from flask import Flask, request
import logging

app = Flask(__name__)
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

@app.route("/rolldice")
def roll_dice():
    player = request.args.get('player', default=None, type=str)
    result = str(roll())
    if player:

>         logger.warning("%s esta jogando os dados: %s", player, result)

Reviewer: ```suggestion
        logger.warning("%s está jogando o dado: %s", player, result)
```

Author: @maryliag se adicionar assento o logs fica bugado console. Por isso removi acentuação.

Reviewer: É que eu assumi que quem está vendo essa documentação fala português e geralmente usa UTF-8. Mas por garantir tudo bem ficar sem mesmo

    else:

>         logger.warning("Jogador anonimo esta jogando os dados: %s", result)

Reviewer: ```suggestion
        logger.warning("Jogador anônimo está jogando o dado: %s", result)
```

    return result

def roll():
    return randint(1, 6)
```

> Execute o aplicativo com o seguinte comando e acesse

Reviewer 2: ```suggestion
Execute a aplicação utilizando o comando abaixo e acesse
```

<http://localhost:8080/rolldice> no seu navegador para garantir que está
funcionando.

```sh
flask run -p 8080
```

## Instrumentação {#instrumentation}

> A instrumentação sem código gerará dados de telemetria para você. Existem várias

Reviewer: ```suggestion
A instrumentação sem código gerará dados de telemetria em seu nome. Existem várias
```

> opções que você pode seguir, cobertas em mais detalhes em

Reviewer: ```suggestion
opções que você pode seguir, abordadas em mais detalhes em
```

> [Instrumentação sem Código](/docs/zero-code/python/). Aqui usaremos o agente

Reviewer 2: ```suggestion
[Instrumentação sem código](/docs/zero-code/python/). Aqui usaremos o agente
```

`opentelemetry-instrument`.

> Instale o pacote `opentelemetry-distro`, que contém a API, SDK do OpenTelemetry
> e também as ferramentas `opentelemetry-bootstrap` e `opentelemetry-instrument`
> que você usará a seguir.

Reviewer 2: Acha que a sugestão abaixo fica mais fluída?

```suggestion
Instale o pacote `opentelemetry-distro`, que contém a API e SDK do OpenTelemetry, além das ferramentas `opentelemetry-bootstrap` e `opentelemetry-instrument`
que serão utilizadas a seguir.
```

```shell
pip install opentelemetry-distro
```

Execute o comando `opentelemetry-bootstrap`:

```shell
opentelemetry-bootstrap -a install
```

Isso instalará a instrumentação do Flask.

> ## Execute o aplicativo instrumentado {#run-the-instrumented-app}

Reviewer 2: ```suggestion
## Execute a aplicação instrumentada {#run-the-instrumented-app}
```

> Agora você pode executar seu aplicativo instrumentado com
> `opentelemetry-instrument` e fazer com que ele imprima no console por enquanto:

Reviewer 2: ```suggestion
Agora, você poderá executar a sua aplicação instrumentada com
`opentelemetry-instrument` e fazer com que os dados sejam emitidos no console:
```

```shell
export OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true
opentelemetry-instrument \
    --traces_exporter console \
    --metrics_exporter console \
    --logs_exporter console \
    --service_name dice-server \
    flask run -p 8080
```

Acesse <http://localhost:8080/rolldice> no seu navegador e recarregue a página

> algumas vezes. Depois de um tempo, você deverá ver os trechos impressos no

Reviewer 2: ```suggestion
algumas vezes. Depois de um tempo, você deverá ver os trechos exibidos no
```

```json
{
    "name": "/rolldice",
    "context": {
        "trace_id": "0xdb1fc322141e64eb84f5bd8a8b1c6d1f",
        "span_id": "0x5c2b0f851030d17d",
        "trace_state": "[]"
    },
    "kind": "SpanKind.SERVER",
    "parent_id": null,
    "start_time": "2023-10-10T08:14:32.630332Z",
    "end_time": "2023-10-10T08:14:32.631523Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "http.method": "GET",
        "http.server_name": "127.0.0.1",
        "http.scheme": "http",
        "net.host.port": 8080,
        "http.host": "localhost:8080",
        "http.target": "/rolldice?rolls=12",
        "net.peer.ip": "127.0.0.1",
        "http.user_agent": "curl/8.1.2",
        "net.peer.port": 58419,
        "http.flavor": "1.1",
        "http.route": "/rolldice",
        "http.status_code": 200
    },
    "events": [],
    "links": [],
    "resource": {
        "attributes": {
            "telemetry.sdk.language": "python",
            "telemetry.sdk.name": "opentelemetry",
            "telemetry.sdk.version": "1.17.0",
            "service.name": "dice-server",
            "telemetry.auto.version": "0.38b0"
        },
        "schema_url": ""
    }
}
{

>     "body": "Jogador anonimo esta jogando os dados: 3",

Reviewer 2: ```suggestion
    "body": "Jogador anônimo esta jogando os dados: 3",
```

Author: Se adicionar assento o logs fica bugado no console devido à codificação nem todo mundo usa oUTF-8. Por isso removi acentuação.

>     "body": "Jogador anonimo esta jogando os dados: 3",

Reviewer: ```suggestion
    "body": "Jogador anônimo está jogando o dado: 3",
```

    "severity_number": "<SeverityNumber.WARN: 13>",
    "severity_text": "WARNING",
    "attributes": {
        "otelSpanID": "5c2b0f851030d17d",
        "otelTraceID": "db1fc322141e64eb84f5bd8a8b1c6d1f",
        "otelServiceName": "dice-server"
    },
    "timestamp": "2023-10-10T08:14:32.631195Z",
    "trace_id": "0xdb1fc322141e64eb84f5bd8a8b1c6d1f",
    "span_id": "0x5c2b0f851030d17d",
    "trace_flags": 1,
    "resource": "BoundedAttributes({'telemetry.sdk.language': 'python', 'telemetry.sdk.name': 'opentelemetry', 'telemetry.sdk.version': '1.17.0', 'service.name': 'dice-server', 'telemetry.auto.version': '0.38b0'}, maxlen=None)"
}
```

</details>

> O trecho gerado rastreia a duração de uma solicitação para a rota `/rolldice`.

Reviewer 2: Usando como referência a mesma tradução em NodeJS: https://opentelemetry.io/pt/docs/languages/js/getting-started/nodejs/

```suggestion
O trecho gerado rastreia o tempo de vida de uma requisição para a rota `/rolldice`.
```

A linha de log emitida durante a solicitação contém o mesmo ID de rastro e ID de
trecho e é exportada para o console via o exportador de logs.

> Envie mais algumas solicitações para o endpoint e em seguida, espere um pouco ou

Reviewer 2: ```suggestion
Envie mais algumas solicitações para esta rota e em seguida, espere um pouco ou
```

> termine o aplicativo e você verá métricas na saída do console, como o seguinte:

Reviewer: ```suggestion
pare a execução da aplicação e você verá métricas na saída do console, como o seguinte:
```

<details>
<summary>Ver exemplo de saída</summary>

```json
{
  "resource_metrics": [
    {
      "resource": {
        "attributes": {
          "service.name": "unknown_service",
          "telemetry.auto.version": "0.34b0",
          "telemetry.sdk.language": "python",
          "telemetry.sdk.name": "opentelemetry",
          "telemetry.sdk.version": "1.13.0"
        },
        "schema_url": ""
      },
      "schema_url": "",
      "scope_metrics": [
        {
          "metrics": [
            {
              "data": {
                "aggregation_temporality": 2,
                "data_points": [
                  {
                    "attributes": {
                      "http.flavor": "1.1",
                      "http.host": "localhost:5000",
                      "http.method": "GET",
                      "http.scheme": "http",
                      "http.server_name": "127.0.0.1"
                    },
                    "start_time_unix_nano": 1666077040061693305,
                    "time_unix_nano": 1666077098181107419,
                    "value": 0
                  }
                ],
                "is_monotonic": false
              },

>               "description": "mede o número de solicitações HTTP simultâneas que estão atualmente em andamento",

Reviewer 2: ```suggestion
              "description": "mede o número de requisições HTTP simultâneas que estão atualmente em andamento",
```

              "name": "http.server.active_requests",
              "unit": "requests"
            },
            {
              "data": {
                "aggregation_temporality": 2,
                "data_points": [
                  {
                    "attributes": {
                      "http.flavor": "1.1",
                      "http.host": "localhost:5000",
                      "http.method": "GET",
                      "http.scheme": "http",
                      "http.server_name": "127.0.0.1",
                      "http.status_code": 200,
                      "net.host.port": 5000
                    },
                    "bucket_counts": [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
                    "count": 1,
                    "explicit_bounds": [
                      0, 5, 10, 25, 50, 75, 100, 250, 500, 1000
                    ],
                    "max": 1,
                    "min": 1,
                    "start_time_unix_nano": 1666077040063027610,
                    "sum": 1,
                    "time_unix_nano": 1666077098181107419
                  }
                ]
              },

>               "description": "mede a duração da solicitação HTTP recebida",

Reviewer 2: ```suggestion
              "description": "mede a duração da requisição HTTP recebida",
```

              "name": "http.server.duration",
              "unit": "ms"
            }
          ],
          "schema_url": "",
          "scope": {
            "name": "opentelemetry.instrumentation.flask",
            "schema_url": "",
            "version": "0.34b0"
          }
        }
      ]
    }
  ]
}
```

</details>

## Adicione instrumentação manual à instrumentação automática {#add-manual-instrumentation-to-automatic-instrumentation}

A instrumentação automática captura telemetria nas bordas dos seus sistemas,

> como solicitações HTTP de entrada e saída, mas não captura o que está

Reviewer 2: ```suggestion
como requisições HTTP de entrada e saída, mas não captura o que está
```

> acontecendo no seu aplicativo. Para isso, você precisará escrever alguma

Reviewer 2: ```suggestion
acontecendo na sua aplicação. Para isso, você precisará escrever alguma
```

[instrumentação manual](../instrumentation/). Aqui está como você pode
facilmente vincular a instrumentação manual com a instrumentação automática.

### Rastros {#traces}

Primeiro, modifique `app.py` para incluir código que inicializa um rastreador e

> utiliza para criar um rastro que é filho do que é gerado automaticamente:

Reviewer: ```suggestion
o utiliza para criar um rastro que é filho do que é gerado automaticamente:
```

```python
from random import randint
from flask import Flask

from opentelemetry import trace

# Adquira um rastreador
tracer = trace.get_tracer("diceroller.tracer")

app = Flask(__name__)

@app.route("/rolldice")
def roll_dice():
    return str(roll())

def roll():
    # Isso cria um novo trecho que é filho do atual
    with tracer.start_as_current_span("roll") as rollspan:
        res = randint(1, 6)
        rollspan.set_attribute("roll.value", res)
        return res
```

> Agora execute o aplicativo novamente:

Reviewer 2: ```suggestion
Agora, execute a aplicação novamente:
```

```shell
export OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true
opentelemetry-instrument \
    --traces_exporter console \
    --metrics_exporter console \
    --logs_exporter console \
    --service_name dice-server \
    flask run -p 8080
```

> Quando você enviar uma solicitação ao servidor, verá dois trechos no rastro

Reviewer 2: ```suggestion
Quando você enviar uma requisição ao servidor, verá dois trechos no rastro
```

    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "http.method": "GET",
        "http.server_name": "127.0.0.1",
        "http.scheme": "http",
        "net.host.port": 8080,
        "http.host": "localhost:8080",
        "http.target": "/rolldice?rolls=12",
        "net.peer.ip": "127.0.0.1",
        "http.user_agent": "curl/8.1.2",
        "net.peer.port": 58485,
        "http.flavor": "1.1",
        "http.route": "/rolldice",
        "http.status_code": 200
    },
    "events": [],
    "links": [],
    "resource": {
        "attributes": {
            "telemetry.sdk.language": "python",
            "telemetry.sdk.name": "opentelemetry",
            "telemetry.sdk.version": "1.17.0",
            "service.name": "dice-server",
            "telemetry.auto.version": "0.38b0"
        },
        "schema_url": ""
    }
}
```

</details>

O `parent_id` de `roll` é o mesmo que o `span_id` para `/rolldice`, indicando
uma relação pai-filho!

### Métricas {#metrics}

Agora modifique `app.py` para incluir código que inicializa um medidor e o usa
para criar um instrumento contador que conta o número de jogadas para cada valor
de jogada possível:

```python

> # Há mais imports necessários agora

Reviewer: ```suggestion
# Estas são as declarações de _imports_ necessários 
```

from opentelemetry import trace
from opentelemetry import metrics

from random import randint
from flask import Flask, request
import logging

# Adquira um rastreador
tracer = trace.get_tracer("diceroller.tracer")
# Adquira um medidor.
meter = metrics.get_meter("diceroller.meter")

# Agora crie um instrumento contador para fazer medições
roll_counter = meter.create_counter(
    "dice.rolls",
    description="O número de jogadas por valor de jogada",
)

app = Flask(__name__)
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

@app.route("/rolldice")
def roll_dice():
    # Isso cria um novo trecho que é filho do atual
    with tracer.start_as_current_span("roll") as roll_span:
        player = request.args.get('player', default = None, type = str)
        result = str(roll())
        roll_span.set_attribute("roll.value", result)
        # Isso adiciona 1 ao contador para o valor de jogada dado
        roll_counter.add(1, {"roll.value": result})
        if player:

>             logger.warn("{} esta jogando os dados: {}", player, result)

Reviewer: ```suggestion
            logger.warn("{} está jogando o dado: {}", player, result)
```

        else:

>             logger.warn("Jogador anonimo esta jogando os dados: %s", result)

Reviewer: ```suggestion
            logger.warn("Jogador anônimo está jogando o dado: %s", result)
```

        return result

def roll():
    return randint(1, 6)
```

> Agora execute o aplicativo novamente:

Reviewer 2: ```suggestion
Agora, execute a aplicação novamente:
```

```shell
export OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true
opentelemetry-instrument \
    --traces_exporter console \
    --metrics_exporter console \
    --logs_exporter console \
    --service_name dice-server \
    flask run -p 8080
```

> Quando você enviar uma solicitação para o servidor, verá a métrica do contador

Reviewer 2: ```suggestion
Quando você enviar uma requisição para o servidor, verá a métrica do contador
```

> de jogadas emitida para o console, com contagens separadas para cada valor de
> jogada:

Reviewer 2: ```suggestion
de jogadas emitida para o console, com contagens de valor separadas para cada jogada:
```

<details>
<summary>Ver exemplo de saída</summary>

```json
{
  "resource_metrics": [
    {
      "resource": {
        "attributes": {
          "telemetry.sdk.language": "python",
          "telemetry.sdk.name": "opentelemetry",
          "telemetry.sdk.version": "1.17.0",
          "service.name": "dice-server",
          "telemetry.auto.version": "0.38b0"
        },
        "schema_url": ""
      },
      "scope_metrics": [
        {
          "scope": {
            "name": "opentelemetry.instrumentation.flask",
            "version": "0.38b0",
            "schema_url": ""
          },
          "metrics": [
            {
              "name": "http.server.active_requests",

>               "description": "mede o número de solicitações HTTP simultâneas que estão atualmente em andamento",

Reviewer 2: ```suggestion
              "description": "mede o número de requisições HTTP simultâneas que estão atualmente em andamento",
```

              "unit": "requests",
              "data": {
                "data_points": [
                  {
                    "attributes": {
                      "http.method": "GET",
                      "http.host": "localhost:8080",
                      "http.scheme": "http",
                      "http.flavor": "1.1",
                      "http.server_name": "127.0.0.1"
                    },
                    "start_time_unix_nano": 1696926005694857000,
                    "time_unix_nano": 1696926063549782000,
                    "value": 0
                  }
                ],
                "aggregation_temporality": 2,
                "is_monotonic": false
              }
            },
            {
              "name": "http.server.duration",

>               "description": "mede a duração da solicitação HTTP recebida",

Reviewer 2: ```suggestion
              "description": "mede a duração da requisição HTTP recebida",
```

                    "value": 1
                  },
                  {
                    "attributes": {
                      "roll.value": "1"
                    },
                    "start_time_unix_nano": 1696926005695491000,
                    "time_unix_nano": 1696926063549782000,
                    "value": 1
                  },
                  {
                    "attributes": {
                      "roll.value": "3"
                    },
                    "start_time_unix_nano": 1696926005695491000,
                    "time_unix_nano": 1696926063549782000,
                    "value": 1
                  },
                  {
                    "attributes": {
                      "roll.value": "4"
                    },
                    "start_time_unix_nano": 1696926005695491000,
                    "time_unix_nano": 1696926063549782000,
                    "value": 1
                  }
                ],
                "aggregation_temporality": 2,
                "is_monotonic": true
              }
            }
          ],
          "schema_url": ""
        }
      ],
      "schema_url": ""
    }
  ]
}
```

</details>

> ## Envie telemetria para um Coletor OpenTelemetry {#send-telemetry-to-an-opentelemetry-collector}

Reviewer 2: ```suggestion
## Envie telemetria para o OpenTelemetry Collector {#send-telemetry-to-an-opentelemetry-collector}
```

O [OpenTelemetry Collector](/docs/collector/) é um componente crítico da maioria

> das implantações em produção. Alguns exemplos de quando é benéfico usar um
> coletor:

Reviewer 2: Ficaria mais fluído dessa maneira? Além disso, faz sentido utilizar o termo `Collector`, como componente? 

```suggestion
das implantações em produção. Alguns exemplos de quando é benéfico utilizar um Collector:
```

> - Um único ponto de telemetria compartilhado por vários serviços, para reduzir a

Reviewer: ```suggestion
- Um único coletor de telemetria compartilhado por vários serviços, para reduzir a
```

  sobrecarga de troca de exportadores

> - Agregando rastros entre vários serviços, executados em vários hosts

Reviewer 2: Pensei em utilizarmos "instâncias", mas não sei se essa é a melhor tradução para "hosts" 🤔 

```suggestion
- Agregando rastros entre vários serviços, executados em várias instâncias
```

> - Um local central para processar rastros antes de exportá-los para um backend

Reviewer 2: ```suggestion
- Um local central para processar rastros antes de exportá-los para um _backend_
```

A menos que você tenha apenas um único serviço ou esteja experimentando, você

> desejará usar um coletor em implantações de produção.

Reviewer 2: ```suggestion
desejará usar um Collector em implantações de produção.
```

> ### Configure e execute um coletor local {#configure-and-run-a-local-collector}

Reviewer 2: ```suggestion
### Configure e execute um Collector local {#configure-and-run-a-local-collector}
```

> Primeiro, salve o seguinte código de configuração do coletor em um arquivo no

Reviewer 2: ```suggestion
Primeiro, salve o seguinte código de configuração do Collector em um arquivo no
```

diretório `/tmp/`:

```yaml
# /tmp/otel-collector-config.yaml
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:
        endpoint: 0.0.0.0:4318
exporters:
  # NOTA: Antes da v0.86.0 use `logging` em vez de `debug`.
  debug:
    verbosity: detailed
processors:
  batch:
service:
  pipelines:
    traces:
      receivers: [otlp]
      exporters: [debug]
      processors: [batch]
    metrics:
      receivers: [otlp]
      exporters: [debug]
      processors: [batch]
    logs:
      receivers: [otlp]
      exporters: [debug]
      processors: [batch]
```

> Em seguida, execute o comando docker para adquirir e executar o coletor com base

Reviewer 2: ```suggestion
Em seguida, execute o comando docker para adquirir e executar o Collector com base
```

nesta configuração:

```shell
docker run -p 4317:4317 \
    -v /tmp/otel-collector-config.yaml:/etc/otel-collector-config.yaml \
    otel/opentelemetry-collector:latest \
    --config=/etc/otel-collector-config.yaml
```

> Agora você terá uma instância do coletor em execução localmente, ouvindo na
> porta 4317.

Reviewer 2: ```suggestion
Agora, você terá uma instância do Collector em execução localmente, ouvindo na
porta 4317.
```

> ### Modifique o comando para exportar trechos e métricas via OTLP {#modify-the-command-to-export- spans-and-metrics-via-otlp}

Reviewer: ```suggestion
### Modifique o comando para exportar trechos e métricas via OTLP {#modify-the-command-to-export-spans-and-metrics-via-otlp}
```

O próximo passo é modificar o comando para enviar trechos e métricas para o
coletor via OTLP em vez do console.

Para fazer isso, instale o pacote de exportador OTLP:

```shell
pip install opentelemetry-exporter-otlp
```

O agente `opentelemetry-instrument` detectará o pacote que você acabou de
instalar e usará a exportação OTLP na próxima vez que for executado.

### Execute o aplicativo {#run-the-application}

> Execute o aplicativo como antes, mas não exporte para o console:

Reviewer 2: ```suggestion
Execute a aplicação como antes, mas não exporte para o console:
```

```shell
export OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true
opentelemetry-instrument --logs_exporter otlp flask run -p 8080
```

Por padrão, `opentelemetry-instrument` exporta rastros e métricas via OTLP/gRPC

> e os enviará para `localhost:4317`, que é onde o coletor está ouvindo.

Reviewer 2: ```suggestion
e os enviará para `localhost:4317`, endereço onde o Collector está ouvindo.
```

     -> http.flavor: STRING(1.1)
     -> http.route: STRING(/rolldice)
     -> http.status_code: INT(200)

2022-06-09T20:43:40.025Z        INFO    debugexporter/debug_exporter.go:56  MetricsExporter {"#metrics": 1}
2022-06-09T20:43:40.025Z        DEBUG   debugexporter/debug_exporter.go:66  ResourceMetrics #0
Resource labels:
     -> telemetry.sdk.language: STRING(python)
     -> telemetry.sdk.name: STRING(opentelemetry)
     -> telemetry.sdk.version: STRING(1.12.0rc1)
     -> telemetry.auto.version: STRING(0.31b0)
     -> service.name: STRING(unknown_service)
InstrumentationLibraryMetrics #0
InstrumentationLibrary app
Metric #0
Descriptor:
     -> Name: roll_counter
     -> Description: O número de jogadas por valor de jogada
     -> Unit:
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: AGGREGATION_TEMPORALITY_CUMULATIVE
NumberDataPoints #0
Data point attributes:
     -> roll.value: INT(5)
StartTimestamp: 2022-06-09 20:43:37.390226915 +0000 UTC
Timestamp: 2022-06-09 20:43:39.848587966 +0000 UTC
Value: 1
```

</details>

## Próximos passos {#next-steps}

Existem várias opções disponíveis para instrumentação automática em Python. Veja

> [Instrumentação sem Código](/docs/zero-code/python/) para aprender sobre e como

Reviewer 2: ```suggestion
[Instrumentação sem código](/docs/zero-code/python/) para aprender sobre e como
```

configurá-las.

Há muito mais na instrumentação manual do que apenas criar um trecho filho. Para
aprender detalhes sobre a inicialização da instrumentação manual e muitas outras
partes da API do OpenTelemetry que você pode usar, veja
[Instrumentação Manual](../instrumentation/).

Existem várias opções para exportar seus dados de telemetria com OpenTelemetry.

> Para aprender como exportar seus dados para um backend preferido, veja

Reviewer 2: ```suggestion
Para aprender como exportar seus dados para um _backend_ preferido, veja
```
